mirrord/agent/src/runtime.rs

```diff
@@ -137,24 +161,43 @@ pub(crate) struct ContainerdContainer {
 async fn connect_and_find_container(
     container_id: String,
     sock_path: impl AsRef<std::path::Path>,
-) -> Result<TasksClient<Channel>> {
+) -> Result<Channel> {
     let channel = connect(sock_path).await?;
-    let mut client = TasksClient::new(channel);
+    let mut client = TasksClient::new(channel.clone());
     let request = GetRequest {
         container_id,
         ..Default::default()
     };
     let request = with_namespace!(request, DEFAULT_CONTAINERD_NAMESPACE);
     client.get(request).await?;
-    Ok(client)
+    Ok(channel)
 }
 
+/// Translate ToIter<String> of "K=V" to HashMap.
+fn parse_raw_env<'a, T: IntoIterator<Item = &'a String>>(raw: T) -> HashMap<String, String> {
+    raw.into_iter()
+        .map(|key_and_value| key_and_value.splitn(2, '=').collect::<Vec<_>>())
+        // [["DB", "foo.db"], ["PORT", "99"], ["HOST"], ["PATH", "/fake"]]
+        .filter_map(
+            |mut keys_and_values| match (keys_and_values.pop(), keys_and_values.pop()) {
+                (Some(value), Some(key)) => Some((key.to_string(), value.to_string())),
+                _ => None,
+            },
```

Comment:
What do you think about something like this:
```suggestion
        .map(|key_and_value| key_and_value.splitn(2, '='))
        // [["DB", "foo.db"], ["PORT", "99"], ["HOST"], ["PATH", "/fake"]]
        .map(|pair_iter| (pair_iter.next(), pair_iter.next()))
        .filter(|(key, value)| value.is_some())
```

Reply:
it makes the pair_iter `Option<K>, Option<V>` afaik